### PR TITLE
Add test for issue9899 (rules for missing/new data members)

### PR DESCRIPTION
**Pull request by pcanal**
[See #9913](https://github.com/root-project/root/pull/9913)

### CODE REVIEW

File: root/io/datamodelevolution/00/test2.cxx [hunk 1]
```diff
@@ -39,17 +39,32 @@ void test2() {
    c->a.Print();
 
    TTree *tree; f->GetObject("tree",tree);
+   cout << "TTree::Scan object branch\n";
```
Comment (Axel-Naumann) [inline root/io/datamodelevolution/00/test2.cxx:42]: Maybe you need a `flush` here to fix the sporadic test failure? Kinda strange...

**Review by Axel-Naumann — APPROVED** [review]
But see test failures.

File: root/io/datamodelevolution/00/test2.cxx (hunk 1 above)
Comment (pcanal) [reply root/io/datamodelevolution/00/test2.cxx:42]: fixed.

**Comment by pcanal** [comment]
The failures are unrelated

**Pull request merged**